### PR TITLE
ActiveSync: can be controlled by module constraint

### PR DESCRIPTION
**Pull request by javierag**
Added 'ActiveSync' module constraint. If constraint is not
honored the ActiveSync request returns a no autorized error (403).

NEWS line suggestion:
Added 'ActiveSync' module constraint.

### CODE REVIEW

**Comment by sixstone-qq** [comment]
Missing `NEWS` line suggestion.

File: ActiveSync/SOGoActiveSyncDispatcher.m [hunk 1]
```diff
@@ -2761,6 +2762,13 @@ - (NSException *) dispatchRequest: (id) theRequest
     
   ASSIGN(context, theContext);
 
+  activeUser = [context activeUser];
+  if (![activeUser canAccessModule: @"ActiveSync"]) {
```
Comment (blaxter) [inline ActiveSync/SOGoActiveSyncDispatcher.m]: Please follow the current code style

File: SoObjects/SOGo/SOGoUserManager.m [hunk 2]
```diff
@@ -768,14 +776,17 @@ - (void) _fillContactInfosForUser: (NSMutableDictionary *) currentUser
...
-          if (!access)
-            [currentUser setObject: [NSNumber numberWithBool: NO]
-                            forKey: @"MailAccess"];
+
+          enumerator = [access_types_list objectEnumerator];
+          while ((access_type = [enumerator nextObject]) != nil)
+            {
+              access = [[userEntry objectForKey: access_type] boolValue];
+              if (!access)
+                {
```
Comment (blaxter) [inline SoObjects/SOGo/SOGoUserManager.m]: The current code style is to not put brackets here (simple if's with only one line)

File: SoObjects/SOGo/SOGoUserManager.m [hunk 3]
```diff
@@ -738,10 +744,12 @@ - (void) _fillContactInfosForUser: (NSMutableDictionary *) currentUser
   c_imaplogin = nil;
   c_sievehostname = nil;
 
-  [currentUser setObject: [NSNumber numberWithBool: YES]
-                  forKey: @"CalendarAccess"];
-  [currentUser setObject: [NSNumber numberWithBool: YES]
-                  forKey: @"MailAccess"];
+  enumerator = [access_types_list objectEnumerator];
+  while ((access_type = [enumerator nextObject]) != nil)
+    {
```
Comment (blaxter) [inline SoObjects/SOGo/SOGoUserManager.m]: I'd remove the brackets to follow current code style

**Comment by blaxter** [comment]
Not tested but lgtm :+1:

**Comment by javierag** [comment]
Applied Blaxter's corrections

**Comment by sixstone-qq** [comment]
Are the other options documented somewhere? I didn't find it. However, I'd like to test it somehow.

**Comment by javierag** [comment]
When the other parts are ready the deploy will be generate the correct sogo.conf file, however if you want to check this independently you have to alter /etc/sogo/sogo.conf to add to the domain modules constraints :

ModulesConstraints = {
  ActiveSync = {
     msExchOmaAdminWirelessEnable = 0;
   };
};

**Comment by papajulio** [comment]
Tested and working as expected :+1:

**Pull request merged**